src/persist-client/src/impl/state.rs

```diff
@@ -84,6 +84,14 @@ where
             }));
         }
 
+        // Only allow empty batches at empty time intervals.
```

Comment:
nit: this reads to me backwards from what you meant. counterproposal: "If the time interval is empty, then there must also be no updates"

Reply:
done.